circuits/transfer/src/execute/variants.rs

```diff
@@ -76,7 +76,7 @@ macro_rules! execute_circuit_variant {
                     * JubJubScalar::zero())
                     + (GENERATOR_NUMS_EXTENDED * blinding_factor);
 
-                let fee = fee.gas_limit;
+                let fee = fee.gas_limit * fee.gas_price;
```

Comment:
I'm feeling like we should probably have a method to returns that in `fee`, but let's see when we're refactoring the "currency".

Reply:
Maybe it could be included in phoenix core as part of the `Fee` struct. In any case, I agree the logic should probably be encapsulated somewhere.